api4/oauth_test.go

```diff
@@ -154,6 +154,27 @@ func TestUpdateOAuthApp(t *testing.T) {
 	oapp.Id = "junk"
 	_, resp = AdminClient.UpdateOAuthApp(oapp)
 	CheckBadRequestStatus(t, resp)
+
+	th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ServiceSettings.EnableOAuthServiceProvider = true })
```

Comment:
A question: on line 88, the same thing happens with the same set of permissions

```
th.AddPermissionToRole(model.PERMISSION_MANAGE_OAUTH.Id, model.SYSTEM_USER_ROLE_ID)
th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ServiceSettings.EnableOAuthServiceProvider = true })
```

And the Update is able to change IsTrusted from false to true successfully. But here in this test, we do the same thing and it cannot. Does somehow the first test has the `PERMISSION_MANAGE_SYSTEM` permission which somehow gets disabled later ?

Reply:
The call to Update on line 109 is made through the AdminClient, otherwise it should fail to change.

Reply:
That was it ! Thanks :)